src/module.cpp

```diff
@@ -262,7 +262,6 @@ instruction_ref module::insert_instruction(instruction_ref ins,
                                            std::vector<instruction_ref> args)
 {
     assert(has_instruction(ins) or is_end(ins, this->end()));
-    assert(not starts_with(op.name(), "@"));
```

Comment:
Allow parameter but not anything else. 

perhaps we should have something like `replace_return`. `replace_parameter` to make it  explicit and not allow `builitin` inside `insert_instruction`.  @pfultz2  what are your thoughts ?

Reply:
Need to create `replace_parameter_shape()` function. 

Names of the parameter needs to remain constant.

Reply:
Used remove instruction function to remove unused parameters, as Paul suggested.